org.eclipse.january.test/src/org/eclipse/january/metadata/MetadataProviderTest.java

```diff
@@ -28,14 +28,14 @@ public void testGetMetadata() throws MetadataException {
 
 		// add IMetadata
 		mdp.addMetadata(new Metadata());
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
+		Assert.assertTrue(mdp.getFirstMetadata(IMetadata.class) instanceof IMetadata);
 		mdp.clearMetadata(null);
 
 		// add sub-class of IMetadata
 		mdp.addMetadata(new InnerMetadata());
-		System.err.println(mdp.getMetadata());
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
-		Assert.assertTrue(mdp.getMetadata() instanceof IInnerMetadata);
+		System.err.println(mdp.getFirstMetadata(IInnerMetadata.class));
```

Comment:
Yeah, it's going to be a little more complicated. Historical metadata users have sub-classed `IMetadata` which is not very scalable and, now, discouraged. As the test implies, they expect both asserts to be true thus the input class can be `IMetadata` or `IInnerMetadata` and return the same thing in this case. The solution is to modify the behaviour of `addMetadata()` and `setMetadata()` (may be just for the `IMetadata` case)